libsolidity/inlineasm/AsmDataForward.h

```diff
@@ -45,11 +45,13 @@ struct If;
 struct Switch;
 struct Case;
 struct ForLoop;
+struct ExpressionStatement;
 struct Block;
 
 struct TypedName;
 
-using Statement = boost::variant<Instruction, Literal, Label, StackAssignment, Identifier, Assignment, FunctionCall, FunctionalInstruction, VariableDeclaration, FunctionDefinition, If, Switch, ForLoop, Block>;
+using Expression = boost::variant<FunctionalInstruction, FunctionCall, Identifier, Literal>;
+using Statement = boost::variant<ExpressionStatement, Instruction, Label, StackAssignment, Assignment, VariableDeclaration, FunctionDefinition, If, Switch, ForLoop, Block>;
```

Comment:
Why isn't an `Instruction` a statement (apart from the fact it makes it easier this way in the rest of the code)?

Reply:
Instruction here is just the plain instruction, not the FunctionalInstruction, i.e. the one that takes its arguments from the stack. This is exactly the distinction that the split into expression and statement would like to make visible.